src/app/components/popup-host/popup-host.component.ts

```diff
@@ -1,20 +1,50 @@
-import {ChangeDetectionStrategy, Component, HostBinding, Input, TemplateRef} from '@angular/core';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    HostBinding,
+    OnDestroy,
+    OnInit,
+    ViewChild,
+    ViewContainerRef,
+} from '@angular/core';
+import {Subject, takeUntil} from 'rxjs';
+import {PopupHostService} from './popup-host.service';
 
 @Component({
     selector: 'app-popup-host',
     templateUrl: './popup-host.component.html',
     styleUrls: ['./popup-host.component.css'],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class PopupHostComponent {
-    @Input() template: TemplateRef<unknown> | null = null;
+export class PopupHostComponent implements OnInit, OnDestroy {
+    private readonly template$ = this.popupHostService.currentTemplate$;
+    private readonly destroy$ = new Subject<void>();
 
-    @HostBinding('class.empty')
-    get isTemplateNullable(): boolean {
-        return !this.template;
+    @ViewChild('container', {read: ViewContainerRef, static: true})
+    container: ViewContainerRef | null = null;
+
+    constructor(readonly popupHostService: PopupHostService) {}
+
+    ngOnInit(): void {
+        this.template$.pipe(takeUntil(this.destroy$)).subscribe(view => {
+            this.container?.clear();
+
+            if (view) {
+                this.shouldShowPopup = false;
+                // this.container?.createEmbeddedView(template);
+                this.container?.insert(view);
+
+                return;
+            }
+
+            this.shouldShowPopup = true;
```

Comment:
Можно упростить и уйти от копипасты

```suggestion
            this.shouldShowPopup = Boolean(view);
            
            if (view) {
                // this.container?.createEmbeddedView(template);
                this.container?.insert(view);

                return;
            }
```

Reply:
исправил